gensim/matutils.py

```diff
@@ -382,6 +383,110 @@ def cossim(vec1, vec2):
     return result
 
 
+def isbow(vec):
+    """
+    Checks if vector passed is in bag of words representation or not.
+    """
+    if scipy.sparse.issparse(vec):
+        vec = vec.todense().tolist()
+    for item in vec:
+        if type(item) is int or type(item) is long or type(item) is float:
+            return False
+        if len(item) != 2:
```

Comment:
See `utils.is_corpus` for a better 2-tuple check: https://github.com/piskvorky/gensim/blob/develop/gensim/utils.py#L591

It also includes a little heuristic for the two tuple types inside (=the two elements must be int-able and float-able).